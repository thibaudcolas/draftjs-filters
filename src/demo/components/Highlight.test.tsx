import { shallow } from "enzyme"
import Highlight from "./Highlight"

describe("Highlight", () => {
  it("renders", () => {
    expect(shallow(<Highlight value="" />)).toMatchSnapshot()
  })

  it("onCopy", () => {
    document.execCommand = jest.fn()
    const wrapper = shallow(<Highlight value="" />)

    wrapper.find("button").simulate("click")

    expect(document.execCommand).toHaveBeenCalledWith("copy")
  })
})
