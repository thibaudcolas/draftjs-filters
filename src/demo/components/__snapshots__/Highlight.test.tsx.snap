// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight renders 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <button
    onClick={[Function]}
    style={
      Object {
        "position": "absolute",
        "right": "1rem",
      }
    }
  >
    Copy
  </button>
  <textarea
    readOnly={true}
    style={
      Object {
        "minHeight": "100px",
        "resize": "vertical",
        "width": "100%",
      }
    }
    value=""
  />
</div>
`;
